dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/SdkFeature.kt

```diff
@@ -156,6 +159,36 @@ internal class SdkFeature(
         )
     }
 
+    override fun <T : Any> writeToDataStore(
+        dataStoreFileName: String,
+        featureName: String,
+        serializer: Serializer<T>,
+        data: T
+    ) {
+        coreFeature.dataStoreHandler.write(
+            dataStoreFileName = dataStoreFileName,
+            featureName = featureName,
+            serializer = serializer,
+            data = data
+        )
+    }
+
+    override fun <T : Any> readFromDataStore(
```

Comment:
if we use the persistence thread I think this operation could contain a callback as a lambda which will return the `T?` value